var/spack/repos/builtin/packages/sundials/package.py

```diff
@@ -107,18 +105,8 @@ class Sundials(CMakePackage):
             description='Enable Fortran 2003 interface')
 
     # Examples
-    variant('examples-c',       default=True,
-            description='Enable C examples')
-    variant('examples-cxx',     default=False,
-            description='Enable C++ examples')
-    variant('examples-f77',     default=True,
-            description='Enable Fortran 77 examples')
-    variant('examples-f90',     default=False,
-            description='Enable Fortran 90 examples')
-    variant('examples-f2003',   default=False,
-            description='Enable Fortran 2003 examples')
-    variant('examples-cuda',    default=False,
-            description='Enable CUDA examples')
+    variant('examples',         default=True,
+            description='Enable examples')
```

Comment:
@balos1 this change triggers a build error with xsdk
```
/usr/bin/ld: cannot find -lsundials_nvecopenmp
```
I guess I could use ^sundials~examples in xsdk

[spack-build-out.txt](https://github.com/spack/spack/files/6240785/spack-build-out.txt)

Reply:
Looks like the issue is with the ARKODE SuperLU_DIST example. It links to the `sundials_nvecopenmp` target but that vector is not used in the example.

@balay try is patch and let me know if it resolves the problem [remove-links-to-OpenMP-vector.patch.txt](https://github.com/spack/spack/files/6240926/remove-links-to-OpenMP-vector.patch.txt)

Reply:
Thanks! sundials@develop build works  with this patch (via xsdk@develop)

Reply:
BTW this issue comes with xsdk@0.6.0 [using sundials@5.5.0] - and perhaps other versions aswell.
The patch gets xsdk@0.6.0 build working.

Reply:
Thanks Satish, I suspect this issue has existed since that example was added but didn't show up until now as the C++ examples were off by default until this change to a single examples variant. We'll need to check what version this was added and make sure the patch can be applied from there until the next release.

Reply:
@gardner48 @balos1 Checking whats the best way to resolve this wrt xsdk@0.6.0 - which uses sundials@5.5.0

Should I use sundials~examples  or should I be adding this patch to sundials@5.5.0 [and perhaps other releases that need it]?

Reply:
Also - is this fix not yet in sundials@develop? [as a build of sundials@develop is accepting this patch - and chugging along]

```
==> Installing sundials-develop-h2bh3cpq5mp5nfmdvtcwbf5kjlibcvhb
==> No binary for sundials-develop-h2bh3cpq5mp5nfmdvtcwbf5kjlibcvhb found: installing from source
==> Warning: Missing a source id for sundials@develop
==> Warning: Missing a source id for hypre@develop
==> Warning: Missing a source id for superlu-dist@develop
==> Warning: Missing a source id for petsc@main
==> Warning: Missing a source id for trilinos@develop
/Users/xsdk/spack-xsdk/lib/spack/spack/architecture.py:205: UserWarning: microarchitecture specific optimizations are not supported yet on mixed compiler toolchains [check apple-clang@12.0.0 for further details]
  warnings.warn(msg.format(compiler))
==> Applied patch /Users/xsdk/spack-xsdk/var/spack/repos/builtin/packages/sundials/remove-links-to-OpenMP-vector.patch
```

Reply:
@balay This is not fixed in sundials@develop yet which is currently the same as sundials@5.7.0. The next sundials release will include the above patch but in the mean time the patch should work for sundials@5.5.0:5.7.0.